### PR TITLE
fix #74

### LINKED ISSUE #74: Cli tool generates wrong enum encoding
It looks like enum generation is broken in the cli tool. The "enc" array in generated code contains "undefined" for enum, which leads to errors like "TypeError: Cannot read property 'encodingLength' of undefined".
See https://gist.github.com/vkosh/d8bc50fd17cad8e51e9c2b8725655017#file-test-proto-js-L32

### PR DESCRIPTION
**Pull request by soyuka**

### CODE REVIEW

File: bin.js [hunk 1]
```diff
@@ -122,7 +122,7 @@ function compile () {
       messages[name].dependencies.forEach(function (e, i, enc) {
         var name = encodings.name(e)
         if (name) name = 'encodings.' + name
-        else name = e.name
+        else name = 'encodings.varint'
```
Comment (soyuka) [inline bin.js]: In fact enums are encoders and need to be added too

Comment (mafintosh) [reply bin.js]: did you mean to close this? :)

Comment (soyuka) [reply bin.js]: Mhh I actually am not sure :laughing: I made some tests and it looks like now it's working but I can't decode -_-.

**Comment by soyuka** [comment]
I've made a small test on the output, should I add it?

```shell
node bin.js test/test.proto -o test/output.js
```

```javascript
var test = require('./test/output')

var buf = test.Defaults.encode({foo1: test.FOO.A, foos: [test.FOO.B]})

console.log(buf)

var dec = test.Defaults.decode(buf)

console.log(dec)
```

**Comment by mafintosh** [comment]
@soyuka let me add a way to test the compiled protobuf against all our current tests instead :)

**Comment by soyuka** [comment]
oups just added the test also

**Comment by mafintosh** [comment]
@soyuka Released in 4.0.3 :) - Changed all current tests to be run against a compiled version of the test schema also.

**Comment by mafintosh** [comment]
For more info about the tests see, https://github.com/mafintosh/protocol-buffers/commit/054bfd1d62f5ed3fac62b644d728a0cfcb882e52

**Pull request merged**